### PR TITLE
[IMP] im_livechat: use guest avatar for live chat visitors 

### PR DESCRIPTION
**Pull request by tsm-odoo**
task-3471571

### CODE REVIEW

File: addons/website_livechat/static/src/core_ui/thread_patch.xml [hunk 1]
```diff
@@ -5,7 +5,7 @@
             <t t-set="visitor" t-value="props.thread.visitor"/>
             <div t-if="visitor and !env.inChatWindow" class="o-website_livechat-VisitorBanner py-4 px-2 d-flex border-bottom">
                 <div class="o-website_livechat-VisitorBanner-sidebar me-2 d-flex justify-content-center">
-                    <img class="rounded o-website_livechat-VisitorBanner-avatar" t-att-src="threadService.avatarUrl(visitor, props.thread)" alt="Avatar"/>
+                    <img class="rounded o-website_livechat-VisitorBanner-avatar" t-att-src="threadService.avatarUrl(props.thread.correspondent, props.thread)" alt="Avatar"/>
```
Comment (alexkuhn) [inline addons/website_livechat/static/src/core_ui/thread_patch.xml:8]: Probably in follow-up PRs to remove `thread.visitor` and rely on `thread.correspondent` all the time

File: addons/im_livechat/static/src/embed/core/thread_model_patch.js [hunk 2]
```diff
@@ -60,6 +60,7 @@ patch(Thread.prototype, {
         super.setup();
         this.chatbotTypingMessage = Record.one("Message");
         this.livechatWelcomeMessage = Record.one("Message");
+        this.operator = Record.one("Persona");
```
Comment (alexkuhn) [inline addons/im_livechat/static/src/embed/core/thread_model_patch.js:63]: Note that you can have proper typing by adding `operator = Persona` in the `@types/models.d.ts`.
See how it's done with `livechatWelcomeMessage` for example!

**Review by alexkuhn — APPROVED** [review]
@robodoo delegate+

**Comment by alexkuhn** [comment]
@robodoo override=ci/security
PY022: routes implementing their own protection (forcing guest, guess token as param)

**Comment by tsm-odoo** [comment]
@robodoo r+

File: addons/im_livechat/static/src/embed/core/thread_service_patch.js [hunk 3]
```diff
@@ -63,20 +63,10 @@ patch(ThreadService.prototype, {
...
-            return `${session.origin}/im_livechat/operator/${
-                author?.id ?? thread.operator.id
-            }/avatar`;
-        } else if (author) {
-            return `${session.origin}/web/image/res.partner/${author.id}/avatar_128`;
-        } else {
-            return `${session.origin}/mail/static/src/img/smiley/avatar.jpg`;
+    avatarUrl(persona, thread) {
+        if (thread.type === "livechat" && persona.eq(thread.operator)) {
+            return url(`/im_livechat/operator/${persona.id}/avatar`);
```
Comment (seb-odoo) [inline addons/im_livechat/static/src/embed/core/thread_service_patch.js:68]: Do we still need the operator specific route? Now there is a guest, the channel route should always work with the token.

Comment (tsm-odoo) [reply addons/im_livechat/static/src/embed/core/thread_service_patch.js:68]: The operator avatar should be available even before the channel (and the guest) are created. So I think this route should be kept.

**Pull request closed without merging**